### PR TITLE
[FIA-2683] Add developer documentation for the Discord integration

### PR DESCRIPTION
**Pull request by gragera**
## Motivation / Description

Add documentation for the discord integration

## Changes introduced

* New discord page
* Add Discord to the list of supported integrations in the "App Development" section

## Linear ticket (if any)

https://linear.app/revenuecat/issue/FIA-2687/update-documentation

### CODE REVIEW

File: docs_source/🔌 Integrations & Events/third-party-integrations/discord.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
+---
+title: Discord
+slug: discord
+excerpt: Send in-app subscription events to Discord
+hidden: true
```
Comment (gragera) [inline docs_source/🔌 Integrations & Events/third-party-integrations/discord.md:5]: @codykerns I have set this to hidden but I don't think is necessary. The frontend changes have not been deployed yet, and I think we can just deploy the frontend and then merge this PR afterwards

File: docs_source/🔌 Integrations & Events/third-party-integrations/discord.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+---
+> 👍 
+> 
+> The Discord integration is available to all users signed up after September '23, the legacy Starter and Pro plans, and Enterprise plans. If you're on a legacy Free plan and want to access this integration, migrate to our new pricing via your [billing settings](https://app.revenuecat.com/settings/billing).
+
+RevenueCat can deliver updates directly to you Discord server when a new event happens in your app.
```
Comment (gragera) [inline docs_source/🔌 Integrations & Events/third-party-integrations/discord.md:11]: @codykerns I'd appreciate some help with the copy here :D

File: docs_source/🔌 Integrations & Events/third-party-integrations/discord.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+# Configure Discord server
+
+You need to configure your Discord server to authorize a webhook to post to your workspace. Please follow this Discord [article](https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks) on their website explaining how to set this up.
+
+# Configure RevenueCat Integration
+
+Once you've set up a webhook in your Discord server. Enter the details into the Discord Integration section of your [dashboard](https://app.revenuecat.com).
+
+<img width="934" alt="Discord integration setup" src="https://github.com/RevenueCat/revenuecat-docs/assets/287089/7764e4de-b9ea-4060-9a59-af0c959bebb0">
```
Comment (gragera) [inline docs_source/🔌 Integrations & Events/third-party-integrations/discord.md:124]: @codykerns Not sure if we should set the width or just remove it

File: docs_source/🔌 Integrations & Events/third-party-integrations/discord.md (hunk 2 above)
Comment (codykerns) [inline docs_source/🔌 Integrations & Events/third-party-integrations/discord.md:11]:
```suggestion
Receive instant updates on your Discord server from RevenueCat whenever a new purchase event occurs in your app.
```

File: docs_source/🔌 Integrations & Events/third-party-integrations/discord.md (hunk 1 above)
Comment (codykerns) [inline docs_source/🔌 Integrations & Events/third-party-integrations/discord.md:5]:
```suggestion
hidden: false
```

File: docs_source/🔌 Integrations & Events/third-party-integrations/discord.md (hunk 3 above)
Comment (codykerns) [reply docs_source/🔌 Integrations & Events/third-party-integrations/discord.md:124]: Maybe we can do it like this format? (found in another doc)

``````

**Review by codykerns — APPROVED** [review]
looks great! just a couple of tweaks

**Review by joshdholtz — APPROVED** [review]
Yesssss... look good to me! 👍

**Pull request merged**